File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> mp1;
        for(const auto&c:magazine){
            mp1[c]++;
        }
        for(const auto&c:ransomNote){
            if(mp1[c]<=0) return false;
            mp1[c]--;
        }
        return true;
    }
};

/*
用map來解決這題mp1
遍歷magazine字串的所有字母並統計字母數量
接著遍歷ramsomNote字串
如果mp1中的字母數量小於等於0
說明字母數量不夠 return false
如果不是就return true
*/

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        #return not (Counter(ransomNote) - Counter(magazine)) 一行解
        records={}
        for c in magazine:
            if c not in records:
                records[c]=0
            records[c]+=1
        for c in ransomNote:
            if c not in records or records[c]<=0:
                return False
            records[c]-=1
        return True